### PR TITLE
Historical Buildings Part I

### PR DESCRIPTION
**Pull request by IhateTrains**
The Queen of Cities first  :P
* Sacred Palace
* Church of the Holy Apostles
* Hagia Sophia

I added French descriptions but they are copied from fr.wikipedia, I don't speak French.

### CODE REVIEW

**Comment by IhateTrains** [comment]
Buildings added so far:
* Sacred Palace
* Hagia Sophia
* Church of the Holy Apostles
* Remnants of Hadrian's Wall
* Remnants of Antonine Wall
* Aurelian Walls
* Colosseum
I'm waiting for a review and further suggestions :)

**Comment by IhateTrains** [comment]
I also received permission from Morloran, author of Unique Buildings to use his work.
 Am I the only one working on this mod at the moment? Are you guys on vacation?

**Comment by TurtleShroom** [comment]
Do you think it will be possible for little pyramids to appear at your capital province if you build a pyramid as a foreign Egyptian Pagan? I once took the "BUILD A PYRAMID" Decision north of Crimea. Sadly, nothing really happened that let me know there was a pyramid.

In the Vanilla rendition, there is actually a Province Event Modifier titled "Pyramid" that decreases unrest. You can get it if you are a Lunatic and certain Event-spawned pets die.

**Comment by IhateTrains** [comment]
If you want to add a building, it would look like this:
https://1drv.ms/t/s!AkkXLHHhb2v-hZ4boYo5SaBT6VYwaw

**Comment by loup99** [comment]
Yes, I was on vacation this week, but now I'm back. 😎

**Comment by loup99** [comment]
My suggestions would be to tie some events and decisions to it. One suggestion we received was about Ad Gefrin (http://www.gefrin.com/) , which I added to the mod previously. Some buildings which are slightly later but could nonetheless be added if you are interested:

- Great Mosque of Damascus or Umayyad Mosque
- Palace of Aachen

**Comment by IhateTrains** [comment]
I wanted to contact TurtleShroom :P Great to see you again, I was starting to worry. Thanks for the suggestions, I will add these buildings and maybe more (we can now use Unique Buildings), e.g. Limes Germanicus.

**Comment by loup99** [comment]
What do you think about using decisions though, like we do for the Circus Maximus?

I think @TurtleShroom is referring to the `build_pyramid `decision in `ACR_egyptian_decisions`. 

I'm going to look at the French descriptions.

File: WTWSMS/localisation/WtWSMS_buildings.csv [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#CODE;ENGLISH;FRENCH;GERMAN;;SPANISH;;;;;;;;;x
```
Comment (loup99) [inline WTWSMS/localisation/WtWSMS_buildings.csv]: The encoding in this file should be in ANSI.

File: WTWSMS/interface/WtWSMS_decisions_icons.gfx [hunk 2]
```diff
@@ -385,4 +393,12 @@ spriteTypes = {
...
 		texturefile = "gfx\\interface\\GFX_create_priscillian_patriarchy.tga"
 	}
+	spriteType = {
+		name = "GFX_peregrinatio"
+		texturefile = "gfx\\interface\\decision_icon_catholic_pilgrimage.dds"
+	}
+	spriteType = {
+		name = "GFX_the_henotikon"
+		texturefile = "gfx\\interface\\decision_icon_write_theory_paper.dds"
+	}
```
Comment (loup99) [inline WTWSMS/interface/WtWSMS_decisions_icons.gfx]: Good ideas on the icons, although the best would be unique ones. These will serve for now though.

File: WTWSMS/localisation/WtWSMS_buildings.csv [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+tp_alexandria_lighthouse_1;Renovated Lighthouse of Alexandria;Phare rénové d'Alexandrie;;;;;;;;;;;;x
+tp_alexandria_lighthouse_1_desc;The Lighthouse of Alexandria has seen better days, but it can be restored to its former greatness.;Le phare d'Alexandrie. Il était considéré dans l'Antiquité comme la septieme des sept merveilles du monde. Il a servi de guide aux marins pendant pres de dix-sept siecles (du IIIe siecle av. J.-C. au XIVe siecle). Sa construction aurait débuté entre -299 et -289 (la date exacte est inconnue) et duré une quinzaine d'années. Les travaux sont commencés par Ptolémée Ier mais il meurt avant la fin du chantier qui est achevé sous le regne de son fils Ptolémée II.;;;;;;;;;;;;x
```
Comment (loup99) [inline WTWSMS/localisation/WtWSMS_buildings.csv]: The issue with using Wikipedia texts is that we can't just copy them in the mod without giving credit. I prefer unique texts and link to the wiki if needed. I can write up something, with inspiration from it, but not copy-paste.

File: WTWSMS/localisation/WtWSMS_buildings.csv (hunk 1 above)
Comment (IhateTrains) [inline WTWSMS/localisation/WtWSMS_buildings.csv]: done

File: WTWSMS/localisation/WtWSMS_buildings.csv (hunk 3 above)
Comment (IhateTrains) [inline WTWSMS/localisation/WtWSMS_buildings.csv]: Shall I replace them or leave this to you?

Comment (loup99) [inline WTWSMS/localisation/WtWSMS_buildings.csv]: My text is usually not the most inspired, but I'm not going to create more work for you, so I will take care of it.

**Comment by loup99** [comment]
Basically, what would be interesting is to add decisions for some of these (all don't need it) and then add the buildings as an effect through `add_building` rather than building them directly in the province.

**Comment by IhateTrains** [comment]
That would be better, e.g. Hagia Sophia could get its own decision and narrative events (Nika riots -> old Hagia Sophia burnt -> Decision to construct a new, enormous Hagia Sophia -> Construction finished, massive prestige and piety boost :P) Sadly, I'm not goot at modding decisions and events and I don't want to break anything.

**Comment by loup99** [comment]
Even with the wiki: http://www.ckiiwiki.com/Decision_modding ? You could also largely copy what we already have, and had in mind for a new suggestion of another building, for the Circus Maximus.

**Comment by IhateTrains** [comment]
I will try tomorrow
I'm not sure but I think #232 was solved by my error fixes.

**Comment by loup99** [comment]
Not sure why my commit closed it? But I nonetheless commited the error fixes directly to be able to test the mod, but didn't merge the whole pull request if you want to try with decisions.

**Comment by IhateTrains** [comment]
If you merge this pull request, please mention Morloran's name and link to his Unique Buildings mod in changelog or credit list.

**Comment by loup99** [comment]
Of course, but you want me to merge it now? I had a few things I would review, and you said the descriptions are placeholders.

**Comment by IhateTrains** [comment]
No, there are many things I want to improve.

**Comment by IhateTrains** [comment]
I think it's ready to be merged. I want add more buildings (among others pyramids and Palace of Aaachen), but this PR is getting too big

File: WTWSMS/common/buildings/WtWSMS_historical_castle.txt [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+		build_time = 1460
+		convert_to_castle = tp_aurelian_walls
+		
+		fort_level = 3.0
+		ai_creation_factor = 86
+
+		extra_tech_building_start = 0.0
+	}
+	
+	ca_carthport_0 = { # Ruined Port of Carthage, credits to Morloran
```
Comment (loup99) [inline WTWSMS/common/buildings/WtWSMS_historical_castle.txt]: The city of Carthage was still important during the period of the mod seat of the praetorian prefecture (later exarchate) of Africa, and it was a port for Roman armies. While it experienced a population decline and much changed when the Vandals took it prior to the start date, ruined is an exaggeration prior to the Islamic conquest and Tunis replacing it.

File: WTWSMS/decisions/WtWSMS_realm_decisions.txt [hunk 5]
```diff
@@ -209,6 +209,33 @@ decisions = {
...
+			NOT = { has_global_flag = hagia_sophia_built }
+			NOT = { has_global_flag = building_hagia_sophia }
+		}
+		allow = {
+			wealth = 1500
+		}
+		effect = {
+			hidden_tooltip = {
+				character_event = { id = imperial_legacy.40 }
+				narrative_event = { id = imperial_legacy.41 days = 2 random = 50 }
```
Comment (loup99) [inline WTWSMS/decisions/WtWSMS_realm_decisions.txt]: Isn't this a bit quick? I would make it into the historical 5 years and 10 months with the 50 random days.

Comment (IhateTrains) [inline WTWSMS/decisions/WtWSMS_realm_decisions.txt]: I forgot to change it 😂 I've set this time to make testing easier

File: WTWSMS/common/buildings/WtWSMS_historical_castle.txt (hunk 4 above)
Comment (IhateTrains) [inline WTWSMS/common/buildings/WtWSMS_historical_castle.txt]: Ok, I will rename it to Port of Carthage and the upgrade to Expanded Port of Carthage

File: WTWSMS/common/buildings/WtWSMS_historical_castle.txt [hunk 6]
```diff
@@ -155,34 +155,13 @@ castle = {
...
-		culture_techpoints = 0.05
-		galleys = 50
-		liege_prestige = 0.5
-		monthly_character_prestige = 0.5
-
-		ai_creation_factor = 0
-		
-		extra_tech_building_start = 8.0
-	}
-
```
Comment (loup99) [inline WTWSMS/common/buildings/WtWSMS_historical_castle.txt]: Why did you delete `carthport_1`? I think it makes sdense to have it, given that the port experienced a decline.

File: WTWSMS/common/buildings/WtWSMS_historical_temple.txt [hunk 7]
```diff
@@ -0,0 +1,183 @@
+temple = {
+
+	tp_hagiasophia = {  # Hagia Sophia
+		potential = {
+			FROMFROM = { title = b_hagiasophia }
+			FROM = { had_global_flag = { flag = hagia_sophia_built days = 0 } }
+		}
```
Comment (loup99) [inline WTWSMS/common/buildings/WtWSMS_historical_temple.txt]: Given that this one is done through decision, it shouldn't be possible to build it directly or at the same time through other means. `always = no` would perhaps be appropriate.

Comment (IhateTrains) [inline WTWSMS/common/buildings/WtWSMS_historical_temple.txt]: I used the had_global_flag to hide the building, when I last tested it couldn't be built directly. I will check again.
`always = no` should be used in potential or trigger?

Comment (loup99) [inline WTWSMS/common/buildings/WtWSMS_historical_temple.txt]: OK, I didn't test it, just that always = no (in potential with maybe a custom tooltip if that exists on buildings, trigger can have the barony) ensures that it doesn't work in any circumstance.

Comment (IhateTrains) [inline WTWSMS/common/buildings/WtWSMS_historical_temple.txt]: I tried `always = no` and Hagia Sophia wasn't visible even after the "Hagia Sophia built" event. I think it should stay as it is IF it doesn't affect the performance.

Comment (loup99) [inline WTWSMS/common/buildings/WtWSMS_historical_temple.txt]: It doesn't, but the important matter is that it is impossible to build it directly.

Comment (IhateTrains) [inline WTWSMS/common/buildings/WtWSMS_historical_temple.txt]: After merging we can ask our testers to check if everything works properly :)

**Comment by loup99** [comment]
With that said, I no longer have any objections. I will merge and at the same time write the new descriptions.
